### ISSUE TITLE
Warn about `**/dir/` Style Exclude/Include Globs

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#16115]
From: https://github.com/Microsoft/vscode/issues/26337#issuecomment-303775870

**TypeScript Version:**  2.3.3

**Code**
Create a `tsconfig.json` that tries to exclude test directories from compilation like so:
```json
{
    "compilerOptions": {
        "outDir": "dist"
    },
    "exclude": [
        "**/test/"
    ]
}
```

**Expected behavior:**
Depends on what TS considers a valid glob:
- If `**/test/` is taken as a valid glob, we should exclude all directories under `test` directories
- If `**/test/` is not a valid glob, we should generate a warning message about this

**Actual behavior:**
Files under `test` directories are still included. No warnings are generated
